koku/masu/test/database/test_aws_report_db_accessor.py

```diff
@@ -259,29 +261,39 @@ def test_bad_sql_execution(self):
             with self.assertRaises(ProgrammingError):
                 accessor._execute_processing_script("masu.database", script_file_path, {})
 
+    @retry(retries=settings.HIVE_PARTITION_DELETE_RETRIES)
+    def delete_ocp_on_aws_hive_partition_by_day(self, *args, **kwargs):
+        return self.accessor.delete_ocp_on_aws_hive_partition_by_day(*args, **kwargs)
+
     @patch("masu.database.aws_report_db_accessor.AWSReportDBAccessor.schema_exists_trino")
     @patch("masu.database.aws_report_db_accessor.AWSReportDBAccessor.table_exists_trino")
-    @patch("masu.database.aws_report_db_accessor.AWSReportDBAccessor._execute_trino_raw_sql_query")
-    def test_delete_ocp_on_aws_hive_partition_by_day(self, mock_trino, mock_table_exist, mock_schema_exists):
+    @patch("masu.database.report_db_accessor_base.trino_db.connect")
+    @patch("time.sleep", return_value=None)
+    def test_delete_ocp_on_aws_hive_partition_by_day(
+        self, mock_sleep, mock_connect, mock_table_exist, mock_schema_exists
+    ):
         """Test that deletions work with retries."""
         mock_schema_exists.return_value = False
         self.accessor.delete_ocp_on_aws_hive_partition_by_day(
             [1], self.aws_provider_uuid, self.ocp_provider_uuid, "2022", "01"
         )
-        mock_trino.assert_not_called()
+        mock_connect.assert_not_called()
+        mock_connect.reset_mock()
 
         mock_schema_exists.return_value = True
-        mock_trino.reset_mock()
-        error = {"errorName": "HIVE_METASTORE_ERROR"}
-        mock_trino.side_effect = TrinoExternalError(error)
-        with self.assertRaises(TrinoExternalError):
+
+        # Setting up the mock so that the cursor execute causes a TrinoExternalError
+        mock_cursor = mock_connect.return_value.cursor.return_value
+        mock_cursor.execute.side_effect = TrinoExternalError({"errorName": "HIVE_METASTORE_ERROR"})
```

Comment:
I don't think this is quite right. I'm actually not sure what this is doing.
Just have `"masu.database.report_db_accessor_base.trino_db.connect` raise an exception. That's enough to verify the retry behavior.
```suggestion
        mock_connect.side_effect = TrinoExternalError({"errorName": "HIVE_METASTORE_ERROR"})
```